app/src/main/java/com/infomaniak/drive/utils/Extensions.kt

```diff
@@ -379,3 +380,40 @@ fun OneTimeWorkRequest.Builder.setExpeditedIfAvailable() = apply {
     if (VERSION.SDK_INT >= VERSION_CODES.S) setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
 }
 //endregion
+
+fun Context.formatShortBinarySize(size: Long, valueOnly: Boolean = false): String {
+
+    fun Long.binaryToDecimal(): Long {
+
+        val binaryUnit = 1_024.0f
+        val decimalUnit = 1_000.0f
+        var units = 0 // BYTE
+        val maxUnits = 5 // KILOBYTE, MEGABYTE, GIGABYTE, TERABYTE, PETABYTE
+        var result = abs(this).toFloat()
+
+        while (result > 900 && units < maxUnits) {
+            units++
+            result /= binaryUnit
+        }
+
+        if (valueOnly) return result.toLong()
+
+        repeat(units) {
+            result *= decimalUnit
+        }
+
+        return result.toLong()
+    }
+
+    val decimalSize = when {
+        VERSION.SDK_INT >= VERSION_CODES.O -> size.binaryToDecimal()
+        valueOnly -> size.binaryToDecimal()
+        else -> size
+    }
+
+    return if (valueOnly) {
+        "$decimalSize"
+    } else {
+        Formatter.formatShortFileSize(this, decimalSize)
+    }
+}
```

Comment:
It could be great to test this method in order to ensure that we have the right output ?

Reply:
Ok, will be done in another PR.